mlrun/feature_store/steps.py

```diff
@@ -505,16 +493,19 @@ def _do_pandas(self, event):
         return event
 
     def _do_spark(self, event):
-        from pyspark.sql.functions import date_format
+        import pyspark.sql.functions
 
         for part in self.parts:
-            if part in self.fstore_date_format_to_spark_date_format:
+            func = part
+            if func == "day_of_year":
+                func = "dayofyear"
+            if func == "day_of_month":
+                func = "dayofmonth"
```

Comment:
```suggestion
            if func in ("day_of_year", "day_of_month"):
                func = func.replace("_", "")
```

Reply:
or even take it to a helper function which will hold all those possible transformations and names

Reply:
I was kind of on the fence about this because who knows what the user might put in there... But sure, why not.

Reply:
Sure, why not.